spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/ApplicationManifestDifferenceSummaryGenerator.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.skipper.server.deployer;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.MapDifference;
+
+import org.springframework.cloud.skipper.domain.deployer.ApplicationManifestDifference;
+
+/**
+ * Generate a summary description of an {@link ApplicationManifestDifference}.
+ * @author Mark Pollack
+ */
+public class ApplicationManifestDifferenceSummaryGenerator {
+
+	private Joiner.MapJoiner mapJoiner = Joiner.on(",").withKeyValueSeparator("=");
+
+	/**
+	 * Generate a textual summary of the ApplicationManifestDifference
+	 * @param applicationManifestDifference the ApplicationManifestDifference
+	 * @return a textual summary
+	 */
+	public String generateSummary(ApplicationManifestDifference applicationManifestDifference) {
+		StringBuffer stringBuffer = new StringBuffer();
+
+		if (applicationManifestDifference.areEqual()) {
+			stringBuffer.append("Existing and Replacing Application Name=["
```

Comment:
Application -> Application(s) and can possibly use `Name` at the end something like:

`Existing and Replacing Applications are equal for the applications named <>`

Reply:
Do you mean `applications named <time>` or `applications named time`.  I ended up with the `Name[time]` approach since otherwise it didn't seem to stand out when looking at the reports.  I added a few println's in the tests that I later removed....

Reply:
Changed to
`Existing and Replacing Applications are equal for Application Name=[time]`
and
`Existing and Replacing are different for Application Name=[time]`